yolo_inference_node:
  ros__parameters:
    model_path: "/workspaces/deep_ros/yolov8m.onnx"
    input_image_topic: "/CAM_FRONT/image_rect_compressed"
    camera_topics: ["/CAM_FRONT/image_rect_compressed", "/CAM_FRONT_LEFT/image_rect_compressed", "/CAM_FRONT_RIGHT/image_rect_compressed"]
    topic_type: "compressed_image"
    queue_size: 10
    input_transport: "compressed"
    input_qos_reliability: "best_effort"
    output_detections_topic: "/detections"
    input_width: 640
    input_height: 640
    use_letterbox: false
    batch_size_limit: 3
    max_batch_latency_ms: 50
    score_threshold: 0.20
    nms_iou_threshold: 0.40
    preferred_provider: "tensorrt"
    device_id: 0
    class_names_path: "/workspaces/deep_ros/deep_yolo_inference/config/coco_classes.txt"
    class_names: [
      "person", "bicycle", "car", "motorbike", "aeroplane", "bus", "train", "truck",
      "boat", "traffic light", "fire hydrant", "stop sign", "parking meter", "bench",
      "bird", "cat", "dog", "horse", "sheep", "cow", "elephant", "bear", "zebra",
      "giraffe", "backpack", "umbrella", "handbag", "tie", "suitcase", "frisbee",
      "skis", "snowboard", "sports ball", "kite", "baseball bat", "baseball glove",
      "skateboard", "surfboard", "tennis racket", "bottle", "wine glass", "cup",
      "fork", "knife", "spoon", "bowl", "banana", "apple", "sandwich", "orange",
      "broccoli", "carrot", "hot dog", "pizza", "donut", "cake", "chair", "sofa",
      "pottedplant", "bed", "diningtable", "toilet", "tvmonitor", "laptop", "mouse",
      "remote", "keyboard", "cell phone", "microwave", "oven", "toaster", "sink",
      "refrigerator", "book", "clock", "vase", "scissors", "teddy bear", "hair drier",
      "toothbrush"
    ]
